internal/utils/retrier.go

```diff
@@ -0,0 +1,61 @@
+package utils
+
+import (
+	"context"
+	"log"
+	"time"
+)
+
+type RetryStrategy string
+
+const (
+	RetryStrategyFixed       RetryStrategy = "fixed"
+	RetryStrategyIncremental               = "incr"
+)
+
+type RetryConfig struct {
+	retryInterval time.Duration
+	retryStrategy RetryStrategy
+	signalChan    chan<- struct{}
+	attempts      uint
+}
+
+func CreateDefaultRetryConfig(signalChan chan<- struct{}) *RetryConfig {
+	return CreateRetryConfig(10*time.Second, RetryStrategyFixed, signalChan)
+}
+
+func CreateRetryConfig(interval time.Duration, strategy RetryStrategy, signalChan chan<- struct{}) *RetryConfig {
+	return &RetryConfig{
+		retryInterval: interval,
+		retryStrategy: strategy,
+		signalChan:    signalChan,
+	}
+}
+
+func (conf *RetryConfig) GetAttempts() uint {
+	return conf.attempts
+}
+
+func (conf *RetryConfig) getNextRetryInterval() time.Duration {
+	attempts := conf.attempts
+	conf.attempts++
+	if conf.retryStrategy == RetryStrategyFixed {
+		return conf.retryInterval
+	}
+	if conf.retryStrategy == RetryStrategyIncremental {
+		return time.Duration(float32(attempts+1)*1.75) * conf.retryInterval
+	}
+	return 0 //unreachable
```

Comment:
Maybe we can make the unreachable panic or return an error so that if somehow it stops being unreachable, we catch it? Or just make the default strategy the default `RetryStrategyFixed` no-if path return?